### ISSUE TITLE
Inconsistency in STAC.yaml and api-spec.md for search endpoint supported methods.

### ISSUE THREAD

**Issue by sharkinsspatial** [radiantearth/stac-spec#711]
The [STAC.yaml](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/STAC.yaml#L137) has been updated to reflect `GET` as an optional method for the `/search` endpoint but the [api-spec.md](https://github.com/radiantearth/stac-spec/blame/dev/api-spec/api-spec.md#L18) seems to only show `POST` support?  I could just be misreading it but should this be updated to reflect the new optional `GET` method?

**Comment by m-mohr** [comment]
The whole section is a bit confusing, I think. The section title is "HTTP Request Methods and Content Types" but the text is basically only describing which content types are supported so I'd propose to change the title to "HTTP Content Types". The text just says that you can use some content types with POST /search, but it doesn't say it's not allowed to use GET. It just doesn't specify a content type for GET /search. Also, I think the chapter should be moved down a bit and be above "Filter Parameters and Fields". And then we should probably add a "Method(s)" column for the "... endpoints" sections. That should clarify it.

**Comment by m-mohr** [comment]
Hey @philvarner, I think you wrote the section "HTTP Request Methods and Content Types". I'm trying to understand the "HTTP Request Methods and Content Types" can you explain it? I don't get it. What POST endpoints is spoken about? What does the bullet point about GET tells me? I think we should clarify that section.

**Comment by philvarner** [comment]
I'll take a look at it

**Comment by philvarner** [comment]
This is a change from my previous view on this -- I think we should now do the following:

* GET w /query params is **required** (maybe some discussion around how this works with a geometry?)
* POST w/ JSON is **optional** but highly recommended (e.g., your service is compliant but pretty trash if you don't implement this)
* POST w/ application/x-www-form-urlencoded is **optional** but recommended. For most web service frameworks, this is either supported by default when GET w/ query params is implemented or it only requires some additional config or a bit of code.
* POST w/ multipart/form-data is not part of this spec. if someone wants to implement that they can, but it is **not** part of the STAC API spec. (for my implementation in Scala and Akka, I couldn't figure out how to get this to work, so I don't support it)

comments? @joshfix

**Comment by joshfix** [comment]
IMHO, I think GET and POST w/JSON should both be required.   I honestly don't care about the other two at all, they can both be optional.  I was able to implement both of the latter, however, with a [single method in Staccato](https://github.com/planetlabs/staccato/blob/0160aab900e758e9244317c7f1925b5dd9a48868/staccato-main/src/main/java/com/planet/staccato/api/ApiApi.java#L36-L38).

**Comment by cholmes** [comment]
I think philosophically with STAC we've settled into requiring as little as possible for a first implementation, and for us 'strongly recommended' does mean 'you should really do this'. 

Though on the other hand I've been comfortable with STAC API requiring a bit more, since those who don't want to implement it can just do the core and point other API's at it.

I think I lean a bit to making it optional + strongly recommended, but I could go either way. And happy for your other recommendations Phil - I've not heard multipart/form-data requested, and agree we shouldn't put it in the spec. Though happy if in the future we have someone make it an extension that they want to promote. And recommending application/x-www-form-urlencoded as an option makes sense, especially if it's easy to implement.

**Comment by cholmes** [comment]
@philvarner - are you up to make a PR on this? Or does anyone else have additional comments? @m-mohr @sharkinsspatial @matthewhanson @scisco

**Comment by m-mohr** [comment]
Well, I'm leaning towards not requiring but recommending, too. My reasoning is simple: openEO implements STAC API, but doesn't expose items and thus can't implement /search and the .../items endpoints in a meaningful way. Not having items is valid in a static catalog and as such should also be valid for the API. Actually, I'm thinking we should expose links with rel type search in the GET / response that point to the search endpoint when implemented.

**Comment by philvarner** [comment]
Thanks for the comments. I'll create a PR today.

**Comment by philvarner** [comment]
@m-mohr Are you saying all of these are only recommended, even GET?  

Are you saying we need to add the "expose links" spec somewhere? I thought having rel=search was already in the API spec? I'm using it here: https://eod-catalog-svc-prod.astraea.earth/api/v2/.

**Comment by m-mohr** [comment]
Yes. Otherwise I'd just implement this endpoint to always returns an empty result set as I'm not exposing items.

I think we should specify somewhere that adding links with rel search should be provided. I couldn't find a place where we do that in STAC API. Good that you have implemented it already!

**Comment by m-mohr** [comment]
@philvarner @joshfix @cholmes and others involved here: While working on PR #731 I realized that the API and README are diverging. The API clearly says: 

> This method [GET /search] is optional, but you MUST implement `POST /search` if you want to implement this method.

and 

> This method [POST /search] is mandatory to implement if `GET /search` is implemented.

I guess we should align the README or decide to define it differently also in the API spec.

My personal opinion is that the API spec should be left as is (/search is optional) and change the README accordingly. Reason as stated above is that implementing STAC API should also be possible if you don't expose items or don't want to implement search, but just allow browsing. I'd be fine to change "optional" to "recommended".

The issue regarding the `search` rel types is not really an issue, it's actually specified, but I clarified it with PR #731.
